### PR TITLE
Spotless and license

### PR DESCRIPTION
**Pull request by cacheung**
## Description

Enable Spotless and checkStyle in CI
Enable javadoc CI
Add License header

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by PravinPK — APPROVED** [review]
Formatting and license changes looks good.

File: code/analytics/src/androidTest/java/com/adobe/marketing/mobile/analytics/internal/MonitorExtension.kt [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
-  Copyright 2022 Adobe. All rights reserved.
```
Comment (kevinlind) [inline code/analytics/src/androidTest/java/com/adobe/marketing/mobile/analytics/internal/MonitorExtension.kt:2]: It's best if we keep the original copyright date.

File: code/analytics/src/test/java/com/adobe/marketing/mobile/analytics/internal/AnalyticsPropertiesTests.kt [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /*
-  Copyright 2022 Adobe. All rights reserved.
```
Comment (kevinlind) [inline code/analytics/src/test/java/com/adobe/marketing/mobile/analytics/internal/AnalyticsPropertiesTests.kt:2]: Can you keep the original copyright date here?

File: code/analytics/src/test/java/com/adobe/marketing/mobile/analytics/internal/AnalyticsRequestSerializerTests.kt [hunk 3]
```diff
@@ -1,5 +1,5 @@
 /*
-  Copyright 2022 Adobe. All rights reserved.
```
Comment (kevinlind) [inline code/analytics/src/test/java/com/adobe/marketing/mobile/analytics/internal/AnalyticsRequestSerializerTests.kt:2]: Can you keep the original copyright date here?

File: code/analytics/src/test/java/com/adobe/marketing/mobile/analytics/internal/AnalyticsStateTests.kt [hunk 4]
```diff
@@ -1,5 +1,5 @@
 /*
-  Copyright 2022 Adobe. All rights reserved.
```
Comment (kevinlind) [inline code/analytics/src/test/java/com/adobe/marketing/mobile/analytics/internal/AnalyticsStateTests.kt:2]: Can you keep the original copyright date here?

File: code/analytics/src/test/java/com/adobe/marketing/mobile/analytics/internal/AnalyticsVersionProviderTests.kt [hunk 5]
```diff
@@ -1,5 +1,5 @@
 /*
-  Copyright 2022 Adobe. All rights reserved.
```
Comment (kevinlind) [inline code/analytics/src/test/java/com/adobe/marketing/mobile/analytics/internal/AnalyticsVersionProviderTests.kt:2]: Can you keep the original copyright date here?

File: code/analytics/src/test/java/com/adobe/marketing/mobile/analytics/internal/TimeZoneTests.kt [hunk 6]
```diff
@@ -1,5 +1,5 @@
 /*
-  Copyright 2022 Adobe. All rights reserved.
```
Comment (kevinlind) [inline code/analytics/src/test/java/com/adobe/marketing/mobile/analytics/internal/TimeZoneTests.kt:2]: Can you keep the original copyright date here?

File: code/analytics/src/androidTest/java/com/adobe/marketing/mobile/analytics/internal/MonitorExtension.kt (hunk 1 above)
Comment (kevinlind) [reply code/analytics/src/androidTest/java/com/adobe/marketing/mobile/analytics/internal/MonitorExtension.kt]: In this case the file already had a copyright notice, so we don't need to change the date. If we copy a file from another repo, we can keep that copyright date. However, if the copied file does not have a copyright notice then we can use the date it was added to the new repo. That's my two cents at least.

**Review by kevinlind — APPROVED** [review]

**Pull request merged**